core/src/main/java/org/apache/gravitino/authorization/OwnerManager.java

```diff
@@ -0,0 +1,192 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.authorization;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import org.apache.gravitino.Entity;
+import org.apache.gravitino.EntityStore;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.SupportsRelationOperations;
+import org.apache.gravitino.exceptions.NoSuchEntityException;
+import org.apache.gravitino.exceptions.NotFoundException;
+import org.apache.gravitino.lock.LockType;
+import org.apache.gravitino.lock.TreeLockUtils;
+import org.apache.gravitino.meta.GroupEntity;
+import org.apache.gravitino.meta.UserEntity;
+import org.apache.gravitino.storage.kv.KvEntityStore;
+import org.apache.gravitino.utils.MetadataObjectUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * OwnerManager is used for manage the owner of metadata object. The user and group don't have an
+ * owner
+ */
+public class OwnerManager {
+  private static final Logger LOG = LoggerFactory.getLogger(OwnerManager.class);
+  private final EntityStore store;
+
+  public OwnerManager(EntityStore store) {
+    if (store instanceof KvEntityStore) {
+      String errorMsg =
+          "OwnerManager cannot run with kv entity store, please configure the entity "
+              + "store to use relational entity store and restart the Gravitino server";
+      LOG.error(errorMsg);
+      throw new RuntimeException(errorMsg);
+    } else if (store instanceof SupportsRelationOperations) {
+      this.store = store;
+    } else {
+      String errorMsg =
+          "OwnerManager currently only supports relational entity store, "
+              + "please configure the entity store to use relational entity store and restart the Gravitino server";
+      LOG.error(errorMsg);
+      throw new RuntimeException(errorMsg);
+    }
+  }
+
+  public void setOwner(
+      String metalake, MetadataObject metadataObject, String ownerName, Owner.Type ownerType) {
+    try {
+      NameIdentifier objectIdent = MetadataObjectUtil.toEntityIdent(metalake, metadataObject);
+      if (ownerType == Owner.Type.USER) {
+        NameIdentifier ownerIdent = AuthorizationUtils.ofUser(metalake, ownerName);
```

Comment:
I would suggest using a different namespace for owner, not reuse "user".

Reply:
Owner isn't an entity. Owner is a relation. The relation will connect the two entities.